viewer/src/views/threejs/cad/discardSector.ts

```diff
@@ -2,24 +2,17 @@
  * Copyright 2020 Cognite AS
  */
 
-import { SectorNode } from './SectorNode';
 import * as THREE from 'three';
 
-export function discardSector(_sectorId: number, sectorNode: SectorNode) {
-  const meshes: THREE.Mesh[] = sectorNode.children.filter(x => x instanceof THREE.Mesh).map(x => x as THREE.Mesh);
+export function discardSector(group: THREE.Group) {
+  const meshes: THREE.Mesh[] = group.children.filter(x => x instanceof THREE.Mesh).map(x => x as THREE.Mesh);
   for (const mesh of meshes) {
     if (mesh.geometry) {
+      console.log("Disposing", mesh.geometry);
       mesh.geometry.dispose();
-    }
-    if (mesh.material && mesh.material instanceof THREE.Material) {
-      mesh.material.dispose();
-    }
-    if (mesh.material && mesh.material instanceof Array) {
-      for (const material of mesh.material) {
-        material.dispose();
-      }
+      // NOTE: Forcefully creating a new reference here to make sure
+      // there are no lingering references to the large geometry buffer
+      mesh.geometry = new THREE.Geometry();
```

Comment:
Create a shared `emptyGeometry` object that we use rather than newing up one every time (but good idea to assign an empty to make the job easier for GC).

Reply:
I agree